test/rate_limiter.js

```diff
@@ -43,7 +43,7 @@ describe ('leaky bucket ratelimiter should', () => {
         let clock = sinon.useFakeTimers(initialDate);
```

Comment:
I was referring to L37 when I said the test is misleading. Let's do a loop to 10 there, like in L30. Also, L31 should probably be an assert.